### ISSUE TITLE
Using %= performs an out-of-place operation instead of in-place

### ISSUE THREAD

**Issue by shz0116** [pytorch/pytorch#49214]
I need to modify a tensor value with % operation inplace. How can I do that ?
```
import torch
a = torch.tensor([11,12,13])
id(a)
a %= 10
id(a)
```
The ids shown are different. If I use "-=" instead of "%=", they will be the same.

cc @ezyang @gchanan @zou3519 @bdhirsh @mruberry @rgommers @heitorschueroff

**Comment by zou3519** [comment]
a.fmod_(10) should give you the answer.

But I think it's weird that %= doesn't modify the tensor in-place

**Comment by zou3519** [comment]
FWIW numpy does modify the tensor in-place:
```
import numpy as np
a = np.array([11,12,13])
print(id(a))
a %= 10
print(id(a))
```
I am not sure what to label this issue as, so I am picking the numpy label

**Comment by heitorschueroff** [comment]
module: numpy is the right label for issues related to NumPy compatibility. I think we should update this to be consistent with NumPy, especially if `-=` is an inplace operation then I would expect all such operations to be inplace.

**Comment by SsnL** [comment]
How is this not a correctness problem?
```py
In [22]: import torch
    ...: a = b = torch.tensor([11,12,13])
    ...: id(a)
    ...: a %= 10
    ...: id(a)
Out[22]: 140322175709824

In [23]:

In [23]: b
Out[23]: tensor([11, 12, 13])

In [24]: a
Out[24]: tensor([1, 2, 3])
```

**Comment by albanD** [comment]
This is not a problem with numpy compat but python compat I think.
These functions are defined by the python standard and should modify the object inplace.

**Comment by ezyang** [comment]
This should be pretty easy to fix, just add an imod implementation.

**Comment by zou3519** [comment]
Unsure if actually BC-breaking (we should be able to get away with this), but we should probably include this in the release notes for our users

**Comment by shz0116** [comment]
By BC-breaking, you mean boundary-check ? I remember I saw this problem when perform % operations using a[0:-1] in a very complicated case.

**Comment by SsnL** [comment]
@shz0116 It means that the behavior will change in a way that old code may not work in the same way in the new version.
